src/app/organizations/settings/delete-organization.component.ts

```diff
@@ -22,10 +25,16 @@ export class DeleteOrganizationComponent {
 
     constructor(private apiService: ApiService, private i18nService: I18nService,
         private toasterService: ToasterService, private userVerificationService: UserVerificationService,
-        private router: Router, private logService: LogService) { }
+        private router: Router, private logService: LogService, private platformUtilsService: PlatformUtilsService) { }
 
     async submit() {
-        const request = await this.userVerificationService.buildRequest(this.masterPassword);
+        let request: SecretVerificationRequest;
+        try {
+            request = await this.userVerificationService.buildRequest(this.masterPassword);
+        } catch (e) {
+            this.platformUtilsService.showToast('error', this.i18nService.t('errorOccurred'), e.message);
+            return;
+        }
```

Comment:
I wonder if we should change `formPromise` to something like:

```ts
this.formPromise = async () => {
    const request = await this.userVerificationService.buildRequest(this.masterPassword);
    return await this.apiService.deleteOrganization(this.organizationId, request);
}
```

Should let us re-use the same try catch. Thoughts?

Reply:
I think this could work, I'll give it a try tomorrow